packages/components/src/Components/InsightsLabel/InsightsLabel.js

```diff
@@ -0,0 +1,55 @@
+import './labels.scss';
+
+import React, { useEffect, useState } from 'react';
+
+import { AngleDoubleDownIcon } from '@patternfly/react-icons';
+import { AngleDoubleUpIcon } from '@patternfly/react-icons';
+import CriticalIcon from './CriticalIcon';
+import { EqualsIcon } from '@patternfly/react-icons';
+import { Label } from '@patternfly/react-core';
+import PropTypes from 'prop-types';
+
+const VALUE_TO_STATE = {
+    1: { icon: <AngleDoubleDownIcon/>, text: 'Low' },
+    2: { icon: <EqualsIcon/>, text: 'Moderate' },
+    3: { icon: <AngleDoubleUpIcon/>, text: 'High' },
+    4: { icon: <CriticalIcon/>, text: 'Critical' }
+};
+
+const InsightsLabel = ({ value, text, hideIcon, labelProps }) => {
+
+    const [ icon, setIcon ] = useState();
+    const [ labelText, setLabelText ] = useState('');
+
+    useEffect(() => {
+        const state = VALUE_TO_STATE?.[value];
+        if (state) {
+            setIcon(state.icon);
+            setLabelText(text || state.text);
+        }
+    }, [ setIcon, value ]);
+
+    return <Label
+        { ...labelProps }
+        className={`ins-c-label-${value}`}
+        icon={!hideIcon && icon}
+    >
+        { labelText }
+    </Label>;
+};
+
+InsightsLabel.propTypes = {
+    value: PropTypes.number,
+    text: PropTypes.string,
```

Comment:
I would use node to allow put there anything. (For example, react intl is using elements to translate texts.)